File: abstractions/typescript/src/authentication/anonymousAuthenticationProvider.ts
import { RequestInformation } from "../requestInformation";
import { AuthenticationProvider } from "./authenticationProvider";

/** This authentication provider does not perform any authentication.   */
export class AnonymousAuthenticationProvider implements AuthenticationProvider {
	public authenticateRequest = (_: RequestInformation): Promise<void> => {
		return Promise.resolve();
	};
}

File: abstractions/go/authentication/anonymous_authentication_provider.go
package authentication

import abs "github.com/microsoft/kiota/abstractions/go"

// AnonymousAuthenticationProvider implements the AuthenticationProvider interface does not perform any authentication.
type AnonymousAuthenticationProvider struct {
}

// AuthenticateRequest is a placeholder method that "authenticates" the RequestInformation instance: no-op.
func (provider *AnonymousAuthenticationProvider) AuthenticateRequest(request abs.RequestInformation) error {
	return nil
}
